app/api/items/add_item.go

```diff
@@ -267,13 +267,15 @@ func (srv *Service) insertItem(store *database.DataStore, user *database.User, f
 	itemMap := formData.ConstructPartialMapForDB("itemWithRequiredType")
 	stringMap := formData.ConstructPartialMapForDB("newItemString")
 
+	service.MustNotBeError(store.Exec("SET FOREIGN_KEY_CHECKS = 0").Error())
 	service.MustNotBeError(store.RetryOnDuplicatePrimaryKeyError(func(s *database.DataStore) error {
 		itemID = s.NewID()
 
 		itemMap["id"] = itemID
 		itemMap["default_language_tag"] = newItemRequest.LanguageTag
 		return s.Items().InsertMap(itemMap)
 	}))
+	service.MustNotBeError(store.Exec("SET FOREIGN_KEY_CHECKS = 0").Error())
```

Comment:
Didn't you want to set to "1" after the block?
BTW, if the goal if to set to 0 before without forgetting to set it back to 1, wouldn't it better to have a fct taking a block as arg? In Golang, this kind of construct are maybe not encouraged.

Reply:
fixed